qiskit/pulse/pulse_lib/pulse.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Any command which implements a transmit signal on a channel."""
+import warnings
+from typing import Callable, Optional
+from abc import ABC, abstractmethod
+
+import numpy as np
+
+from ..channels import PulseChannel
+from ..exceptions import PulseError
+from ..instructions.play import Play
+
+
+class Pulse(ABC):
```

Comment:
This is more akin to a `Type`, do you think it should live in a `types` module along with Kernel and Discriminator?

Reply:
That's an interesting idea, I definitely like the idea of keeping simple, basic, "type-like" objects together and out of the way. For now, I think this does make the most sense. Once `__call__` is removed (since it's sort of un-type-like) and this command->instruction thing is done, we should work on another organizational refactor, including this idea.